script.module.libmediathek3/lib/libmediathek3listing.py

```diff
@@ -89,24 +100,22 @@ def addEntries(l):
 		art['fanart'] = d.get('fanart',d.get('thumb',fanart))
 		art['icon'] = d.get('channelLogo','')
 		liz.setArt(art)
-		if 'type' in d:
-			#if d['type'] == 'clip':
-			if False:
-				xbmc.log('ignoring clip')
-			elif d.get('type',None) == 'video' or d.get('type',None) == 'live' or d.get('type',None) == 'date' or d.get('type',None) == 'clip':
-				xbmcplugin.setContent( handle=int( sys.argv[ 1 ] ), content="episodes" )
-				liz.setProperty('IsPlayable', 'true')
-				lists.append([u,liz,False])
-			elif 'type' in d and d['type'] == 'nextPage':
-				xbmcplugin.setContent( handle=int( sys.argv[ 1 ] ), content="files" )
-				lists.append([u,liz,True])
-			elif d['type'] == 'shows':
-				xbmcplugin.setContent( handle=int( sys.argv[ 1 ] ), content="tvshows" )
-				lists.append([u,liz,True])
-			else:
-				xbmcplugin.setContent( handle=int( sys.argv[ 1 ] ), content="files" )
-				lists.append([u,liz,True])
+		
+		#if d['type'] == 'clip':
+		if False:
+			xbmc.log('ignoring clip')
+		elif d.get('type',None) == 'video' or d.get('type',None) == 'live' or d.get('type',None) == 'date' or d.get('type',None) == 'clip' or d.get('type',None) == 'episode':
+			xbmcplugin.setContent( handle=int( sys.argv[ 1 ] ), content="episodes" )
```

Comment:
you are calling setContent() for every single listitem? You only need to call it once for each folder listing.